### PR TITLE
chore(python): Add various unit tests

### PR DESCRIPTION
**Pull request by zundertj**
And remove unused code.

### CODE REVIEW

File: py-polars/polars/dataframe/frame.py [hunk 1]
```diff
@@ -409,22 +408,6 @@ def _from_pydf(cls, py_df: PyDataFrame) -> Self:
         df._df = py_df
         return df
 
-    @classmethod
-    def _from_dicts(
```
Comment (stinodego) [inline py-polars/polars/dataframe/frame.py:413]: Not 100% I'm willing to part with this code yet. I want to give some thought on these private classmethods - maybe we should get rid of all of them. But let's keep them around until I've decided.

**Review by stinodego — COMMENTED** [review]
Nice, always good to have some more tests! Just two small comments.

File: py-polars/tests/unit/test_cfg.py [hunk 2]
```diff
@@ -577,3 +577,44 @@ def test_config_scope() -> None:
...
+
+def test_config_state_env_only() -> None:
+    pl.Config.set_verbose(False)
+    pl.Config.set_fmt_float("full")
+
+    state_all = pl.Config.state(env_only=False)
+    state_env_only = pl.Config.state(env_only=True)
+    assert len(state_env_only) < len(state_all)
+    assert "set_fmt_float" in state_all
+    assert "set_fmt_float" not in state_env_only
```
[the remark this answers is not in the document]
Comment (zundertj) [reply py-polars/tests/unit/test_cfg.py:595]: There is a auto-use test fixture at the top of this file that does it. I guess I confused you a bit by using the context manager in some of the other tests, but either pattern is safe.

File: py-polars/polars/dataframe/frame.py (hunk 1 above)
Comment (zundertj) [reply py-polars/polars/dataframe/frame.py]: It is superseded by `sequence_to_pydf`,  but I had to triple check myself to ensure it was all good, as it is a bit odd this code is not in use given all the other private methods on DataFrame. Reverted.

File: py-polars/tests/unit/test_cfg.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (stinodego) [reply py-polars/tests/unit/test_cfg.py:595]: Ah you're right, thanks!

**Review by stinodego — APPROVED** [review]
🚀 🚀

**Pull request merged**